cbt_enable_and_snapshot.py

```diff
@@ -1,28 +1,60 @@
 #!/usr/bin/env python
 
-import sys
-import XenAPI
-
-def export_vdi(snapshot_ref):
-    pass
-
+"""
+For a given VDI this script enables cbt, snapshots the VDI and exports the snapshot. This script will only need to be once when first enabling cbt backups and will be the base snapshot. After the VDI has been exported the snapshot data is destroyd to save space on the host.
 
-def todo_download_changed_blocks(bitmap, nbd_info):
-    pass
+example: python cbt_enable_and_snapshot.py -h <host address> -u <host username> -p <host password> -v <vdi uuid> -o <output path of VDI>
 
-vdi_uuid = sys.argv[1]
+Script will then print out the snapshot uuid at the end. VDI is saved to the output path specified
+"""
 
-session = XenAPI.xapi_local()
-session.login_with_password("root", "xenroot", "0.1", "CBT example")
-try:
-    vdi_ref = session.xenapi.VDI.get_by_uuid(vdi_uuid)
-    session.xenapi.VDI.enable_cbt(vdi_ref)
-
-    snapshot_ref = session.xenapi.VDI.snapshot(vdi_ref)
-    export_vdi(snapshot_ref)
-    # Once you are done copying the blocks, delete the snapshot data
-    session.xenapi.VDI.data_destroy(snapshot_ref)
-
-    print session.xenapi.VDI.get_uuid(snapshot_ref)
-finally:
-    session.xenapi.session.logout()
\ No newline at end of file
+import XenAPI
+import shutil
+import urllib3
+import requests
+import argparse
+
+def export_vdi(host, session_id, vdi_uuid, file_format, export_path):
+    url = ('https://%s/export_raw_vdi?session_id=%s&vdi=%s&format=%s'
+           % (host, session_id, vdi_uuid, file_format))
+    with requests.Session() as session:
+        # ToDo: Security - We need to verify the SSL certificate here.
+        # Depends on CP-23051.
+        urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+        request = session.get(url, verify=False, stream=True)
+        with open(export_path, 'wb') as filehandle:
+            shutil.copyfileobj(request.raw, filehandle)
+        request.raise_for_status()
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-ip', '--host-ip', dest='host')
+    parser.add_argument('-u', '--username', dest='username')
+    parser.add_argument('-p', '--password', dest='password')
+    parser.add_argument('-v', '--vdi-uuid', dest='vdi_uuid')
+    parser.add_argument('-o', '--output-path', dest='output_path')
+    args = parser.parse_args()
+
+    host = args.host

```

Comment:
Minor - you could save some lines here, if you were to refer directly to the args instead of copying them.